cfme/web_ui/paginator.py

```diff
@@ -78,28 +86,28 @@ def sort_by(sort):
     sel.select(Select(select), sel.ByText(sort))
 
 
+def _records():
+    matches = re.search(_records_re, _page_nums()).groupdict()
+    return {k: int(v) for k, v in matches.iteritems()}
+
+
 def rec_offset():
     """ Returns the first record offset."""
-    offset = re.search('\((Item|Items)*\s*(\d+)', _page_nums())
-    return offset.groups()[1]
+    return _records()['offset']
 
 
 def rec_end():
     """ Returns the record set index."""
-    offset = re.search('-(\d+)', _page_nums())
-    if offset:
-        return offset.groups()[0]
+    end = _records().get('end')
+    if end:
+        return end
     else:
         return rec_total()
 
 
 def rec_total():
     """ Returns the total number of records."""
-    offset = re.search('(\d+)\)', _page_nums())
-    if offset:
-        return offset.groups()[0]
-    else:
-        return None
+    return _records()['total']
```

Comment:
Again, I didn't put this in for fun. I'm sure it was to address some problem we had with paginators.